pkg/zone/available_services_test.go

```diff
@@ -0,0 +1,218 @@
+		mesh := core_mesh.NewMeshResource()
+		mesh.Spec = &mesh_proto.Mesh{
+			Mtls: &mesh_proto.Mesh_Mtls{
+				EnabledBackend: "builtin",
+				Backends: []*mesh_proto.CertificateAuthorityBackend{
+					{
+						Name: "builtin",
+						Type: "builtin",
+					},
+				},
+			},
+			Routing: &mesh_proto.Routing{
+				ZoneEgress: true,
+			},
+		}
+		Expect(resManager.Create(context.Background(), mesh, store.CreateByKey("mesh1", ""))).To(Succeed())
+		externalService := &core_mesh.ExternalServiceResource{
+			Meta: &test_model.ResourceMeta{
+				Mesh: "mesh1",
+				Name: "es-1",
+			},
+			Spec: &mesh_proto.ExternalService{
+				Networking: &mesh_proto.ExternalService_Networking{
+					Address: "127.0.0.1:80",
+				},
+				Tags: map[string]string{
+					"kuma.io/service":  "httpbin",
+					"version":          "v1",
+					mesh_proto.ZoneTag: "zone",
+				},
+			},
+		}
+		Expect(resManager.Create(context.Background(), externalService, store.CreateByKey("es-1", "mesh1"))).To(Succeed())
+		dp := &core_mesh.DataplaneResource{
+			Spec: &mesh_proto.Dataplane{
+				Networking: &mesh_proto.Dataplane_Networking{
+					Address: "127.0.0.1",
+					Inbound: []*mesh_proto.Dataplane_Networking_Inbound{
+						{
+							Port: 8000,
+							Tags: map[string]string{
+								"kuma.io/service": "backend",
+								"version":         "v1",
+								"region":          "eu",
+							},
+						},
+					},
+				},
+			},
+		}
+		Expect(resManager.Create(context.Background(), dp, store.CreateByKey("dp-1", "mesh1"))).To(Succeed())
+		dp = &core_mesh.DataplaneResource{
+			Spec: &mesh_proto.Dataplane{
+				Networking: &mesh_proto.Dataplane_Networking{
+					Address: "127.0.0.1",
+					Inbound: []*mesh_proto.Dataplane_Networking_Inbound{
+						{
+							Port: 8000,
+							Tags: map[string]string{
+								"kuma.io/service": "web",
+								"version":         "v2",
+								"region":          "us",
+							},
+						},
+					},
+				},
+			},
+		}
+		Expect(resManager.Create(context.Background(), dp, store.CreateByKey("dp-2", "mesh1"))).To(Succeed())
+		dp = &core_mesh.DataplaneResource{
+			Spec: &mesh_proto.Dataplane{
+				Networking: &mesh_proto.Dataplane_Networking{
+					Address: "127.0.0.1",
+					Inbound: []*mesh_proto.Dataplane_Networking_Inbound{
+						{
+							Port: 8000,
+							Tags: map[string]string{
+								"kuma.io/service": "web",
+								"version":         "v2",
+								"region":          "us",
+							},
+						},
+					},
+				},
+			},
+		}
+		Expect(resManager.Create(context.Background(), dp, store.CreateByKey("dp-3", "mesh1"))).To(Succeed())
+
+		tracker, err := zone.NewZoneAvailableServicesTracker(
+			core.Log.WithName("test"),
+			metrics,
+			resManager,
+			meshCache,
+			20*time.Millisecond,
+			nil,
+			"zone",
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		stop := make(chan struct{})
+		done = make(chan struct{})
+		go func() {
+			defer GinkgoRecover()
+			Expect(tracker.Start(stop)).To(Succeed())
+			close(done)
+		}()
+		defer close(stop)
```

Comment:
nit: why do we not start this in Before like in any other component? Do we need to create those resources before we start the component?